desc/objectives/_wrappers.py

```diff
@@ -249,17 +268,33 @@ def jac(self, x):
         Gx = self._objective.jac(xg)
         Fx = align_jacobian(Fx, self._eq_objective, self._objective)
         Gx = align_jacobian(Gx, self._objective, self._eq_objective)
+
+        # projections onto optimization space
         # possibly better way: Gx @ np.eye(Gx.shape[1])[:,self._unfixed_idx] @ self._Z
         Fx_reduced = Fx[:, self._unfixed_idx] @ self._Z
         Gx_reduced = Gx[:, self._unfixed_idx] @ self._Z
-
         Fc = Fx @ dxdc
-        Fx_reduced_inv = jnp.linalg.pinv(Fx_reduced, rcond=1e-6)
-
         Gc = Gx @ dxdc
+
+        # FIXME (@f0uriest): need to import here to avoid circular dependencies
```

Comment:
is this easily fixable?

Reply:
not really, we'd need to move a bunch of stuff around. I started to do it on the `rc/lagrangian` branch but there's no rush.